acceptance/tests/snapshot-agent/snapshot_agent_vault_test.go

```diff
@@ -40,28 +41,136 @@ func TestSnapshotAgent_Vault(t *testing.T) {
+	// KV2 secrets
+	// -------------------------
+	// Gossip key
+	gossipKey, err := vault.GenerateGossipSecret()
+	require.NoError(t, err)
+	gossipSecret := &vault.SaveVaultSecretConfiguration{
+		Path:       "consul/data/secret/gossip",
+		Key:        "gossip",
+		Value:      gossipKey,
+		PolicyName: "gossip",
+	}
+	vault.SaveSecret(t, vaultClient, gossipSecret)
+
+	// License
+	licenseSecret := &vault.SaveVaultSecretConfiguration{
+		Path:       "consul/data/secret/license",
+		Key:        "license",
+		Value:      cfg.EnterpriseLicense,
+		PolicyName: "license",
+	}
 	if cfg.EnableEnterprise {
-		vault.ConfigureEnterpriseLicenseVaultSecret(t, vaultClient, cfg)
+		vault.SaveSecret(t, vaultClient, licenseSecret)
 	}
 
-	bootstrapToken := vault.ConfigureACLTokenVaultSecret(t, vaultClient, "bootstrap")
+	// Bootstrap Token
+	bootstrapToken, err := uuid.GenerateUUID()
+	require.NoError(t, err)
+	bootstrapTokenSecret := &vault.SaveVaultSecretConfiguration{
+		Path:       "consul/data/secret/bootstrap",
+		Key:        "token",
+		Value:      bootstrapToken,
+		PolicyName: "bootstrap",
+	}
+	vault.SaveSecret(t, vaultClient, bootstrapTokenSecret)
 
-	config := generateSnapshotAgentConfig(t, bootstrapToken)
-	vault.ConfigureSnapshotAgentSecret(t, vaultClient, cfg, config)
+	// Snapshot Agent config
+	snapshotAgentConfig := generateSnapshotAgentConfig(t, bootstrapToken)
+	require.NoError(t, err)
+	snapshotAgentConfigSecret := &vault.SaveVaultSecretConfiguration{
+		Path:       "consul/data/secret/snapshot-agent-config",
+		Key:        "config",
+		Value:      snapshotAgentConfig,
+		PolicyName: "snapshot-agent-config",
+	}
+	vault.SaveSecret(t, vaultClient, snapshotAgentConfigSecret)
 
-	serverPolicies := "gossip,connect-ca-dc1,server-cert-dc1,bootstrap-token"
+	// -------------------------
+	// Additional Auth Roles
+	// -------------------------
+	serverPolicies := fmt.Sprintf("%s,%s,%s,%s", gossipSecret.PolicyName, connectCAPolicy, serverPKIConfig.PolicyName, bootstrapTokenSecret.PolicyName)
 	if cfg.EnableEnterprise {
-		serverPolicies += ",license"
+		serverPolicies += fmt.Sprintf(",%s", licenseSecret.PolicyName)
 	}
-	vault.ConfigureKubernetesAuthRole(t, vaultClient, consulReleaseName, ns, "kubernetes", "server", serverPolicies)
-	vault.ConfigureKubernetesAuthRole(t, vaultClient, consulReleaseName, ns, "kubernetes", "client", "gossip")
-	vault.ConfigureKubernetesAuthRole(t, vaultClient, consulReleaseName, ns, "kubernetes", "server-acl-init", "bootstrap-token")
-	vault.ConfigureKubernetesAuthRole(t, vaultClient, consulReleaseName, ns, "kubernetes", "snapshot-agent", "snapshot-agent-config,license")
-	vault.ConfigureConsulCAKubernetesAuthRole(t, vaultClient, ns, "kubernetes")
 
-	vault.ConfigurePKICA(t, vaultClient)
-	certPath := vault.ConfigurePKICertificates(t, vaultClient, consulReleaseName, ns, "dc1", "1h")
+	// server
+	consulServerRole := "server"
+	vault.ConfigureK8SAuthRole(t, vaultClient, &vault.KubernetesAuthRoleConfiguration{
```

Comment:
hmm.. maybe the K8s auth role need not be defined on the struct. it looks like a different one each time.

Reply:
can you elaborate?  do you mean don't have `ConfigureK8SAuthRole()` on the struct?

Reply:
i think it is fine either way. while combing through the tests, I observed that we had 4-6 different structs per test for the Kubernetes authroleconfig so I could understand not wanting to define it on the struct, but I do prefer defining the method on the struct than accepting it as a param. thanks for doing this John!

Reply:
yea, I see what you mean. i agree your suggestion of moving these things to the structs rather than a broad global vault service is nicer.  thanks for the thoughtful feedback!